src/lib/services/workflow-service.ts

```diff
@@ -121,9 +124,24 @@ export async function fetchWorkflowWithEventHistory(
   parameters: GetWorkflowExecutionRequest,
   request = fetch,
 ): Promise<{ workflow: WorkflowExecution; events: HistoryEventWithId[] }> {
+  let port = get(webDecoderPort);
+  // if port rawPayloads = true
+
   const [workflow, events] = await Promise.all([
     fetchWorkflow(parameters, request),
-    fetchEvents(parameters, request).then(toEventHistory),
+    fetchEvents(parameters, request)
+      .then(toEventHistory)
+      .then((events) => {
+        if (port === null) {
+          return events;
+        }
+
+        webDecodeEventPayloads(events, port).catch((error) => {
```

Comment:
Should we be returning this? That promise's `return` will never do anything (it will just `return` into the abyss) and the `catch` will swallow the error.

Reply:
Good call this is definitely erroneous code, I was slinging this snippet around and it is definitely incorrect here.